components/prombench/manifests/benchmark/3_prometheus-test.yaml

```diff
@@ -129,21 +130,25 @@ spec:
         runAsUser: 0
       containers:
       - name: prometheus
-        image: quay.io/prometheus/prometheus:{{ .RELEASE }}
+        image: docker.io/prombench/prometheus-builder:2.0.0
         imagePullPolicy: Always
-        args: [
-          "--config.file=/etc/prometheus/config/prometheus.yaml",
-          "--storage.tsdb.path=/data",
-          "--storage.tsdb.min-block-duration=15m",
-          "--storage.tsdb.max-block-duration=4h"
-        ]
+        # The prometheus-builder takes a while to build and run
+        # so make sure we start it before the release deployment.
+        # Mark it ready only when prometheus is started.
+        # This way we have the least time difference in the scraped metrics.
+        readinessProbe:
+          tcpSocket:
```

Comment:
Won't the following readinessProbe be a good fit? 
```
readinessProbe:
  httpGet:
    path: /-/ready
    port: 9090
  initialDelaySeconds: 30
  periodSeconds: 1
```

Reply:
I guess if for some reason Prom doesn't start properly can have an additional Alert rule for that.